lib/NIF/NifFile.cpp

```diff
@@ -977,8 +977,10 @@ OptResult NifFile::OptimizeFor(OptOptions& options) {
 
 				BSTriShape* bsOptShape = nullptr;
 				//Only converting niTriShape to bsTriShape if required
-                if(options.bsTriShape || shape->HasType<NiTriStrips>() || shape->HasType<bhkMultiSphereShape>()
-		  || options.headParts)
+                		const bool needsOpt = options.bsTriShape || options.headParts || shape->HasType<bhkMultiSphereShape>()
```

Comment:
Shapes (NiShape) can never be bhkMultiSphereShape, NiTriStripsData or NiSkinPartition, so the additional checks won't help. It's also not all NiSkinPartition blocks, but only those with strips in them (they can have both triangles and strips or either).

The entire check should be this:
```cpp
const bool needsOpt = options.bsTriShape || shape->HasType<NiTriStrips>() || hasSkinPartitionWithStrips
```

And it should check for numStrips > 0 in any of the partitions inside the NiSkinPartition block.

```cpp
bool hasSkinPartitionWithStrips = false;

auto skinInst = hdr.GetBlock<NiSkinInstance>(shape->GetSkinInstanceRef());
if (skinInst) {
	skinPart = hdr.GetBlock<NiSkinPartition>(skinInst->GetSkinPartitionRef());
	if (skinPart) {
		// loop through skinPart->partitions and check for numStrips > 0
		//hasSkinPartitionWithStrips = true;
	}
}
```